### PR TITLE
chore: upgrade chokidar to v4

### PR DESCRIPTION
**Pull request by talentlessguy**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

extra 11 dependencies

## What is the new behavior?

11 dependencies less, better TypeeScript support

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

E2E tests pass. maybe worth adding a `is-glob` to filter out non-glob inputs and not wrap them in a glob?

### CODE REVIEW

File: packages/graphql/lib/graphql-definitions.factory.ts [hunk 1]
```diff
@@ -45,7 +46,7 @@ export class GraphQLDefinitionsFactory {
         'GraphQL factory is watching your files...',
         isDebugEnabled,
       );
-      const watcher = chokidar.watch(options.typePaths);
+      const watcher = chokidar.watch(await glob(options.typePaths));
```
Comment (kamilmysliwiec) [inline packages/graphql/lib/graphql-definitions.factory.ts:49]: How is this change related to the PR?

Comment (talentlessguy) [reply packages/graphql/lib/graphql-definitions.factory.ts:49]: chokidar@4 removed support for globs

**Comment by kamilmysliwiec** [comment]
LGTM
